pandas/io/formats/xml.py

```diff
@@ -247,17 +252,38 @@ def other_namespaces(self) -> dict:
 
         return nmsp_dict
 
-    def build_attribs(self) -> None:
+    def build_attribs(self, d: dict[str, Any], elem_row: Any) -> Any:
         """
         Create attributes of row.
 
         This method adds attributes using attr_cols to row element and
         works with tuples for multindex or hierarchical columns.
         """
 
-        raise AbstractMethodError(self)
+        if not self.attr_cols:
+            return elem_row
+
+        for col in self.attr_cols:
+            attr_name = self._get_flat_col_name(col)
+            try:
+                val = None if isna(d[col]) else str(d[col])
+                if val is not None:
+                    elem_row.attrib[attr_name] = val
```

Comment:
should this be copied?

Reply:
Maybe avoid `val`? (This change came on last recent `to_xml` PR).

```python
if not isna(d[col]):
    elem_row.attrib[attr_name] = str(d[col])
```

Reply:
Yep thx, this sounds good